web/src/app/webview/src/contextManager.ts

```diff
@@ -183,4 +188,9 @@ export default class ContextManager extends ContextManagerBase<WebviewConfigurat
 
     return activatingKeyboard;
   }
+
+  public resetContext(): void {
+    super.resetContext();
+    this._rawContext.saveState();
```

Comment:
So, how does this small change fix the issue?

https://github.com/keymanapp/keyman/blob/a92d475c1f66c27cb16811f7a6943395917fd88f/web/src/app/webview/src/contextManager.ts#L22-L23

https://github.com/keymanapp/keyman/blob/a92d475c1f66c27cb16811f7a6943395917fd88f/web/src/app/webview/src/contextManager.ts#L39

We use the most recent version of the context given to us by the host app and do a diff from that.  The key, of course, is that it's kept up to date.

The Android codepaths (`updateKMText`, `updateKMSelectionRange`) both use `resetContext` when a significant change occurs.  Saving the state after a context reset also... just makes sense to me, thematically.

Reply:
The 17.0 versions of this error resulted from using a `savedState` that was created at the old caret position, not the new one, because the code added by this PR, to update `savedState`, was missing.